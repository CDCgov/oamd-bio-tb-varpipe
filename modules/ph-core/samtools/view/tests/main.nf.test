nextflow_process {

    name "Test Process SAMTOOLS_VIEW"
    script "../main.nf"
    process "SAMTOOLS_VIEW"

    tag "modules"
    tag "modules_phcore"
    tag "samtools"
    tag "samtools/view"

    test("bam") {

        when {
            process {
                """
                // Prepare input data
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/bam/test.paired_end.bam', checkIfExists: true),
                    []
                ])
                input[1] = [[],[]]
                input[2] = []
                input[3] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("bam_stub") {

        options "-stub"
        config "./bam_index.config"

        when {
            process {
                """
                // Prepare input data
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/bam/test.paired_end.bam', checkIfExists: true),
                    []
                ])
                input[1] = [[],[]]
                input[2] = []
                input[3] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
