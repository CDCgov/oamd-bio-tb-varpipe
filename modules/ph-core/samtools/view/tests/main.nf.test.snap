{
    "bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam:md5,acd020df9e7f06a80fcfc48cb3744f4c"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    "versions.yml:md5,176db5ec46b965219604bcdbb3ef9e07"
                ],
                "bai": [
                    
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam:md5,acd020df9e7f06a80fcfc48cb3744f4c"
                    ]
                ],
                "crai": [
                    
                ],
                "cram": [
                    
                ],
                "csi": [
                    
                ],
                "sam": [
                    
                ],
                "unselected": [
                    
                ],
                "unselected_index": [
                    
                ],
                "versions": [
                    "versions.yml:md5,176db5ec46b965219604bcdbb3ef9e07"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.1"
        },
        "timestamp": "2025-05-19T11:05:14.155253548"
    },
    "bam_stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam.csi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    "versions.yml:md5,176db5ec46b965219604bcdbb3ef9e07"
                ],
                "bai": [
                    
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "crai": [
                    
                ],
                "cram": [
                    
                ],
                "csi": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam.csi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "sam": [
                    
                ],
                "unselected": [
                    
                ],
                "unselected_index": [
                    
                ],
                "versions": [
                    "versions.yml:md5,176db5ec46b965219604bcdbb3ef9e07"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.1"
        },
        "timestamp": "2025-05-19T11:05:21.125796598"
    }
}
